### DISCUSSION TITLE
support SeaweedFS as storage backend

### DISCUSSION THREAD

**Discussion by ghost** [juicedata/juicefs#4194] [General]
**What would you like to be added**:
support SeaweedFS as storage backend

**Why is this needed**:
SeaweedFS is very optimized for small files and it has a different protocal compared to s3.

**Comment by SandyXSD** [comment]
Any issue when you use it with the S3 protocol?

**Comment by ghost** [comment]
s3 of SeaweedFS requires setting up [Filer](https://github.com/seaweedfs/seaweedfs/wiki/Filer-Stores), which works like an alternative of juicefs.

seeweedFS-like system will generate a key for the uploaded file instead of using the user providied key like s3.

SeaweedFS is (kind of) similar to the block device mode in juicefs for acceleratingsmall files , but SeaweedFS does not require shared block-device and it's better for a clould-native enviroment.

**Comment by davies** [comment]
It's very difficult to use SeaweedFS without the filer under current architecture, so we won't put it int he roadmap.
